sdk/test/metrics/sum_aggregation_test.cc

```diff
@@ -0,0 +1,208 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+#include "opentelemetry/common/macros.h"
+#include "opentelemetry/sdk/metrics/data/point_data.h"
+#include "opentelemetry/sdk/metrics/meter.h"
+#include "opentelemetry/sdk/metrics/meter_context.h"
+#include "opentelemetry/sdk/metrics/meter_provider.h"
+#include "opentelemetry/sdk/metrics/metric_reader.h"
+#include "opentelemetry/sdk/metrics/push_metric_exporter.h"
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry;
+using namespace opentelemetry::sdk::instrumentationscope;
+using namespace opentelemetry::sdk::metrics;
+
+class MockMetricExporter : public PushMetricExporter
+{
+public:
+  MockMetricExporter() = default;
+  opentelemetry::sdk::common::ExportResult Export(
+      const ResourceMetrics & /* records */) noexcept override
+  {
+    return opentelemetry::sdk::common::ExportResult::kSuccess;
+  }
+
+  AggregationTemporality GetAggregationTemporality(
+      InstrumentType /* instrument_type */) const noexcept override
+  {
+    return AggregationTemporality::kCumulative;
+  }
+
+  bool ForceFlush(std::chrono::microseconds /* timeout */) noexcept override { return true; }
+
+  bool Shutdown(std::chrono::microseconds /* timeout */) noexcept override { return true; }
+};
+
+class MockMetricReader : public MetricReader
+{
+public:
+  MockMetricReader(std::unique_ptr<PushMetricExporter> exporter) : exporter_(std::move(exporter)) {}
+  AggregationTemporality GetAggregationTemporality(
+      InstrumentType instrument_type) const noexcept override
+  {
+    return exporter_->GetAggregationTemporality(instrument_type);
+  }
+  virtual bool OnForceFlush(std::chrono::microseconds /* timeout */) noexcept override
+  {
+    return true;
+  }
+  virtual bool OnShutDown(std::chrono::microseconds /* timeout */) noexcept override
+  {
+    return true;
+  }
+  virtual void OnInitialized() noexcept override {}
+
+private:
+  std::unique_ptr<PushMetricExporter> exporter_;
+};
+
+TEST(HistogramToSum, Double)
+{
+#if OPENTELEMETRY_HAVE_WORKING_REGEX
```

Comment:
nit: shall we move everything under `OPENTELEMETRY_HAVE_WORKING_REGEX` as all the tests need it?

Reply:
Good point, fixed it now.